rocAL/rocAL/source/parameters/parameter_rocal_crop.cpp

```diff
@@ -53,8 +53,8 @@ void RocalCropParam::fill_crop_dims() {
             cropw_arr_val[img_idx] = (crop_w <= in_width[img_idx] && crop_w > 0) ? crop_w : in_width[img_idx];
             croph_arr_val[img_idx] = (crop_h <= in_height[img_idx] && crop_h > 0) ? crop_h : in_height[img_idx];
             if (_is_fixed_crop) {
-                x1_arr_val[img_idx] = static_cast<size_t>(_crop_anchor[0] * (in_width[img_idx] - cropw_arr_val[img_idx]));
-                y1_arr_val[img_idx] = static_cast<size_t>(_crop_anchor[1] * (in_height[img_idx] - croph_arr_val[img_idx]));
+                x1_arr_val[img_idx] = static_cast<size_t>(std::nearbyintf(_crop_anchor[0] * (in_width[img_idx] - cropw_arr_val[img_idx])));
```

Comment:
what is std::nearbyintf (I haven't heard of this)?
We need to use standard std:: round functions instead of internal functions within the library

Reply:
The round() functions round their argument to the nearest integer value in floating-point format, rounding halfway cases away from zero, regardless of the current rounding direction. 
The difference in the handling of these two variants of rounding "to nearest" is apparent when 0.5 is rounded to the next larger (in magnitude) integer (1) when using round(), but it rounds to an even integer (0) when nearbyint() is used. 
DALI uses round() function and it is producing different crop dimensions when compared to torchvision's crop dimensions. We made this change to ensure parity with torchvision's crop dimension calculation.

Reply:
The nearbyintf() functions assumes to set the floating point rounding mode before calling it (FE_TONEAREST). Otherwise it uses the default mode.